shaptools/hana.py

```diff
@@ -447,6 +447,193 @@ def get_sr_status(self):
         status["status"] = SrStatusReturnCode.get(result.returncode, SrStatusReturnCode.UNKNOWN)
         return status
 
+    def set_ini_parameter(
+            self, parameter_list, database, file_name, layer,
+            **kwargs):
+        """
+        set HANA configuration parameters in ini file
+
+        sql syntax:
+        ALTER SYSTEM ALTER CONFIGURATION (<filename>, <layer>[, <layer_name>])
+        SET (<section_name_1>,<parameter_name_1>) = <parameter_value_1>,
+            (<section_name_2>,<parameter_name_2>) = <parameter_value_2>
+        WITH RECONFIGURE
+
+        key_name or user_name/user_password parameters must be used
+        Args:
+            parameter_list(dictionary)
+                section_name (str): section name of parameter in ini file
+                parameter_name (str): name of the parameter to be modified
+                parameter_value (str): the value of the parameter to be set
+            database (str): Database name
+            file_name (str): ini configuration file name
+            layer (str): target layer for the configuration change 'SYSTEM', 'HOST' or 'DATABASE'
+            layer_name (str, optional): target either a tenant name or a target host name
+            reconfig (str, optional): if apply changes to running HANA instance
+            key_name (str, optional): Keystore to connect to sap hana db
+            user_name (str, optional): User to connect to sap hana db
+            user_password (str, optiona): Password to connecto to sap hana db
+        """
+
+        parameter_list_str = ', '.join(
+            "{!s}={!r}".format(key, val)
+            for (key, val) in parameter_list.items())
+
+        key_name = kwargs.get('key_name', None)
+        user_name = kwargs.get('user_name', None)
+        user_password = kwargs.get('user_password', None)
+        hdbsql_cmd = self._hdbsql_connect(
+            key_name=key_name, user_name=user_name, user_password=user_password)
+
+        layer_name = kwargs.get('layer_name', None)
+        if layer in ('HOST', 'DATABASE') and layer_name is not None:
+            layer_name_str = ', \'{}\''.format(layer_name)
+        else:
+            layer_name_str = ''
+
+        reconfig = kwargs.get('reconfig', False)
+        reconfig_option = ' WITH RECONFIGURE' if reconfig else ''
+
+        cmd = ('{hdbsql_cmd} -d {db} '
+               '\"ALTER SYSTEM ALTER CONFIGURATION(\'{file_name}\', \'{layer}\'{layer_name}) SET'
+               '{parameter_list_str}{reconfig};\"'.format(
+                   hdbsql_cmd=hdbsql_cmd, db=database, file_name=file_name, layer=layer,
+                   layer_name=layer_name_str, parameter_list_str=parameter_list_str,
+                   reconfig=reconfig_option))
```

Comment:
The same that b' WITH RECONFIGURE'efore. We could simply add:

```reconfig=' WITH RECONFIGURE' if kwargs.get('reconfig', False) else ''```

And remove the logic above. But again, do it the most comfortable way for you